client/store/types/chapter.ts

```diff
@@ -0,0 +1,45 @@
+import { ThunkAction } from 'redux-thunk';
+import { AppStoreState } from '../reducers';
+
+export const FETCH_START = 'fcc/chapter/CHAPTER_START';
+export const FETCH_SUCCESS = 'fcc/chapter/CHAPTER_SUCCESS';
+export const FETCH_FAIL = 'fcc/chapter/CHAPTER_FAIL';
+
+export interface IChapterModal {
+  id?: string;
```

Comment:
Something we fetch from the backend will always have an ID, this was probably done so we could have it work client side for a form or something like that, but since it's only used for fetching make it mandatory.

Reply:
This was kept, so that same modal can be used for POST body as well, where `id` might be `undefined`/`null`.

If that's not the case, will make `id` as _required_.

Reply:
Yeah, for that case yes, but we're currently not doing that. Anyways just keep it like this for the furure and if we end up not using we can just remove one question mark

Reply:
OR maybe we could have something like this
```
interface Chapter {
  name: string;
}

interface DBChapter extends Chapter {
  id: string;
}

const a: Chapter = { name: "Foobar" };
const b: DBChapter = { id: "something", name: "string" };
```

Reply:
Feels like having an extra interface, just for a single key, is an overkill. Keeping it optional seems more simple. We can move to this architecture, if required later. Hope, you agree with me here. :+1:

Reply:
Will resolve it for now.